ambry-account/src/main/java/com/github/ambry/account/AccountUpdater.java

```diff
@@ -0,0 +1,115 @@
+        String.valueOf(zkConnectionTimeoutMs));
+    helixConfigProps.setProperty(HelixPropertyStoreConfig.HELIX_PROPERTY_STORE_PREFIX + "zk.client.session.timeout.ms",
+        String.valueOf(zkSessionTimeoutMs));
+    helixConfigProps.setProperty(HelixPropertyStoreConfig.HELIX_PROPERTY_STORE_PREFIX + "zk.client.connect.string",
+        zkServer);
+    helixConfigProps.setProperty(HelixPropertyStoreConfig.HELIX_PROPERTY_STORE_PREFIX + "root.path", storePath);
+    vHelixConfigProps = new VerifiableProperties(helixConfigProps);
+    storeConfig = new HelixPropertyStoreConfig(vHelixConfigProps);
+    notifier = new HelixNotifier(storeConfig);
+    accountService =
+        new HelixAccountServiceFactory(vHelixConfigProps, new MetricRegistry(), notifier).getAccountService();
+  }
+
+  /**
+   * Create or update a collection of {@link Account}s in {@code ZooKeeper}. The {@link Account}s are specified
+   * in a file that contains a {@link JSONArray}, with each element be the json format of the {@link Account}s
+   * to create or update.
+   * @param accountJsonFilePath The path of the file that contains a json array of the {@link Account}s to update.
+   * @param zkServer The address of the {@code ZooKeeper} server.
+   * @param storePath The path for {@link org.apache.helix.store.HelixPropertyStore}, which will be used as the
+   *                  root path for both {@link HelixAccountService} and {@link HelixNotifier}.
+   * @param zkConnectionTimeoutMs The timeout in millisecond to connect to the {@code ZooKeeper} server.
+   * @param zkSessionTimeoutMs The timeout in millisecond for a session to the {@code ZooKeeper} server.
+   * @return The number of accounts that have been successfully created or updated. {@code -1} indicates that
+   *         the operation is not successful.
+   * @throws Exception Any exception during the operation.
+   */
+  static int createOrUpdate(String accountJsonFilePath, String zkServer, String storePath,
+      Integer zkConnectionTimeoutMs, Integer zkSessionTimeoutMs) throws Exception {
+    int zkClientConnectionTimeoutMs =
+        zkConnectionTimeoutMs == null ? ZK_CLIENT_CONNECTION_TIMEOUT_MS : zkConnectionTimeoutMs;
+    int zkClientSessionTimeoutMs = zkSessionTimeoutMs == null ? ZK_CLIENT_SESSION_TIMEOUT_MS : zkSessionTimeoutMs;
+    JSONArray accountArray = new JSONArray(Utils.readStringFromFile(accountJsonFilePath));
+    Collection<Account> accounts = new ArrayList<>();
+    for (int i = 0; i < accountArray.length(); i++) {
+      JSONObject accountJson = accountArray.getJSONObject(i);
+      accounts.add(Account.fromJson(accountJson));
+    }
+    int res = -1;
+    if (!hasDuplicateAccountIdOrName(accounts)) {
+      AccountUpdater accountUpdater =
+          new AccountUpdater(zkServer, storePath, zkClientConnectionTimeoutMs, zkClientSessionTimeoutMs);
+      if (accountUpdater.update(accounts)) {
+        res = accountArray.length();
```

Comment:
So the return value is either the number of accounts passed in, or -1? Could this function instead return void and throw an exception instead of returning -1, since we have to catch exceptions in the main method anyways?